CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -524,7 +524,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
                     return true
                 })
                 var values = ckRecord.dictionaryWithValuesForKeys(keys)
-                managedObject.setValuesForKeysWithDictionary(values)
+                managedObject.setValuesForKeysWithDictionary(values) // TODO: it would be safer to filter the keys first to ensure they exist in managedObject. Over time schema changes could result in fields that only on the server - this was covered in one of the videos from WWDC 2014.
```

Comment:
Ok but V1 will never try to set values belonging to any new attributes which it got from V2 because it will only retrieve the values for the keys that belong to the V1 version type of the record from V2. That is what I have implemented in code :)